identity-apps-core/apps/recovery-portal/src/main/webapp/self-registration-username-request.jsp

```diff
@@ -1894,7 +1904,14 @@
                     alphanumeric_username_error_msg.show();
                     alphanumericUsernameField.addClass("error");
 
-                } else if (!/^(?=.*[a-zA-Z])[a-zA-Z0-9]+$/.test(alphanumericUsernameUserInput.value.trim())) {
+                } else if (usernameConfig.enableSpecialCharacters
+                && !/^(?=.*[a-zA-Z])[a-zA-Z0-9!@#$%&'*+\\=?^_`.{|}~-]+$/.test(alphanumericUsernameUserInput.value.trim())) {
```

Comment:
Let's format this condition for better readability. Also let's use a const variable for this regex.

```suggestion
                } else if (
                usernameConfig.enableSpecialCharacters
                && !/^(?=.*[a-zA-Z])[a-zA-Z0-9!@#$%&'*+\\=?^_`{|}~-]+$/.test(alphanumericUsernameUserInput.value.trim())) {
```

Reply:
+1 to use a constant

Reply:
Addressed with https://github.com/wso2/identity-apps/pull/4790/commits/cd880a4bc3295c222c65b70c471cba50594c172f